src/components/calculator.js

```diff
@@ -1,36 +1,43 @@
-import React, { useState } from 'react';
+import React from 'react';
+import calculate from '../logic/calculate';
 
 class Components extends React.Component {
     constructor(props) {
         super(props);
         this.state = {};
+        this.handleClick = this.handleClick.bind(this);
     }
+    handleClick(event) {
+        this.setState((state) => calculate(state, event.target.textContent));
+      }
+
     render() {
+        const { next, total } = this.state;
         return (
             <div className='container'>
                 <form>
-                    <input type="text"/>
+                    <input type="text" value={next || total || 0}/>
                 </form>
                 <div className='keypad'>
-                    <button type='button' className='firstline'>AC</button>
-                    <button type='button' className='firstline'>+/-</button>
-                    <button type='button' className='firstline'>%</button>
-                    <button type='button' className='extra'>&divide;</button>
-                    <button type='button' className='secondline'>7</button>
-                    <button type='button' className='secondline'>8</button>
-                    <button type='button' className='secondline'>9</button>
-                    <button type='button' className='extra'>x</button>
-                    <button type='button' className='thirdline'>4</button>
-                    <button type='button' className='thirdline'>5</button>
-                    <button type='button' className='thirdline'>6</button>
-                    <button type='button' className='extra'>&ndash;</button>
-                    <button type='button' className='fourthline'>1</button>
-                    <button type='button' className='fourthline'>2</button>
-                    <button type='button' className='fourthline'>3</button>
-                    <button type='button' className='extra'>+</button>
-                    <button type='button' className='cero'>0</button>
-                    <button type='button' className='dot'>.</button>
-                    <button type='button' className='equal'>=</button>
+                    <button type='button' className='firstline' onClick={this.handleClick}>AC</button>
+                    <button type='button' className='firstline' onClick={this.handleClick}>+/-</button>
+                    <button type='button' className='firstline' onClick={this.handleClick}>%</button>
```

Comment:
- [ ] **OPTIONAL** In order to learn more about props, I would recommend creating a new button component where you will do the calculation and pass it the state there as a prop from Calculator